src/sentry/issues/occurrence_consumer.py

```diff
@@ -42,6 +46,32 @@ class EventLookupError(Exception):
     pass
 
 
+def create_rate_limit_key(fingerprint: str) -> str:
+    rate_limit_key = f"occurrence_rate_limit:{fingerprint}"
+    return rate_limit_key
+
+
+def is_rate_limited(
+    fingerprint: str,
+) -> bool:
+    rate_limit_enabled = options.get("issues.occurrence-consumer.rate-limit.enabled")
+    if not rate_limit_enabled:
+        return False
+
+    rate_limit_key = create_rate_limit_key(fingerprint)
+    rate_limit_quota = Quota(**options.get("issues.occurrence-consumer.rate-limit.quota"))
+    granted_quota = rate_limiter.check_and_use_quotas(
+        [
+            RequestedQuota(
+                rate_limit_key,
+                1,
+                [rate_limit_quota],
+            )
+        ]
+    )[0]
+    return not granted_quota.granted
```

Comment:
I'd recommend wrapping all of this logic in a try/catch and returning false on exception. There's always a chance we mess up when configuring the option. Might be wise to just wrap the entire function to be safe